### PR TITLE
[#84] Bump nixpkgs

### LINKED ISSUE #84: Bump nixpkgs
# Clarification and motivation
Currently, this repo uses quite old nixpkgs which is incompatible with our other projects, when we try
to use ocaml overlay from this repo. Also it will impove caching.

I've tried to bump nixpkgs but wasn't able to do that quickly because a lot of things
was updated and new weird errors appeared.

# Acceptance criteria
This repo uses nixpkgs from our fork same as most of our other repos.

### PR DESCRIPTION
**Pull request by rvem**
## Description
Problem: Current nixpkgs version is quite old and doesn't use our fork.

Solution: Use revision from our fork that uses most of our repos.

## Related issue(s)

Resolves #84 

Relates #97 

#### Related changes (conditional)

- [x] I checked whether I should update the [README](../tree/master/README.md)

#### Stylistic guide (mandatory)

- [x] My commits comply with [the policy used in Serokell](https://www.notion.so/serokell/Where-and-how-to-commit-your-work-58f8973a4b3142c8abbd2e6fd5b3a08e).

### CODE REVIEW

File: nix/build/hacks.nix [hunk 1]
```diff
@@ -4,23 +4,50 @@
...
   conduit-lwt-unix = osuper.conduit-lwt-unix.versions."2.0.2";
   cohttp-lwt-unix = osuper.cohttp-lwt-unix.versions."2.4.0";
   cohttp-lwt = osuper.cohttp-lwt.versions."2.4.0";
-  macaddr = osuper.macaddr.versions."4.0.0";
+  macaddr = osuper.macaddr.versions."4.0.0".overrideAttrs (_: {
+    minimumOCamlVersion = "4.07";
+  });
 
-  lwt = lwt4;
+  lwt = osuper.lwt.versions."4.2.1";
```
Comment (balsoft) [inline nix/build/hacks.nix:50]: Also override `lwt4` then so that packages from nixpkgs use the same version as packages generated by opam-nix

Comment (rvem) [reply nix/build/hacks.nix:50]: AFAIU, there is no `lwt4` in the nixpkgs anymore

Comment (rvem) [reply nix/build/hacks.nix:50]: ping @balsoft

Comment (gromakovsky) [reply nix/build/hacks.nix:50]: @balsoft

**Review by balsoft — APPROVED** [review]

**Pull request merged**